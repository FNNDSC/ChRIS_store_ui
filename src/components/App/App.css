/* CSS variables */

:root {
  /* PRIMARY COLORS */

  /* black */
  --pf-black-100: #fafafa;

  --pf-black-150: #f5f5f5;

  --pf-black-200: #ededed;

  --pf-black-300: #d1d1d1;

  --pf-black-400: #bbbbbb;

  --pf-black-500: #8b8d8f;

  --pf-black-600: #72767b;

  --pf-black-700: #4d5258;

  --pf-black-800: #393f44;

  --pf-black-900: #292e34;

  --pf-black: #030303;

  /* blue */
  --pf-blue-50: #def3ff;

  --pf-blue-100: #bee1f4;

  --pf-blue-200: #7dc3e8;

  --pf-blue-300: #39a5dc;

  --pf-blue-400: #0088ce;

  --pf-blue-500: #00659c;

  --pf-blue-600: #004368;

  --pf-blue-700: #002235;
}

/* ============================== */
/* ---------- GLOBAL CSS -------- */
/* ============================== */

/* ============================== */
/* ---------- HELPER CSS -------- */
/* ============================== */

.row {
  /* display */
  width: 100%;
  max-width: 1000px;
  margin: 0 auto !important;
}

.text-light {
  /* color */
  color: #8b8d8f;
  color: var(--pf-black-500);
}

.text-white {
  /* color */
  color: #fafafa;
  color: var(--pf-black-100);
}

/* ============================== */
/* ---------- RESPONSIVE -------- */
/* ============================== */

@media (min-width: 768px)  {
  .row:not(.no-flex) { display: flex; }
}
