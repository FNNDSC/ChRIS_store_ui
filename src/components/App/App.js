import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import ConnectedSignIn from '../SignIn/SignIn';
import Router from '../Router/Router';
import ChrisStore from '../../store/ChrisStore';
import './App.css';

// import the patternfly CSS globally
import '../../../node_modules/patternfly/dist/css/patternfly.min.css';
import '../../../node_modules/patternfly/dist/css/patternfly-additions.min.css';
import '../../../node_modules/@patternfly/patternfly/patternfly-no-reset.css';
import '../../../node_modules/@patternfly/patternfly/patternfly-addons.css';

/*
 * The router here serves pages which replace the entire document,
 * whereas '../Router/Router' has a navbar and footer.
 */

const App = () => (
  <ChrisStore.Container>
    <div className="App">
      <BrowserRouter>
        <Switch> 
          <Route path="/signin" component={ConnectedSignIn} />
          <Route path="/" component={Router} />
        </Switch>
      </BrowserRouter>
    </div>
  </ChrisStore.Container>
);

export default App;
