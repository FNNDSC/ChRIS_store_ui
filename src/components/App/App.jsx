import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import ConnectedSignIn from '../SignIn/SignIn';
import Router from '../Router/Router';
import ChrisStore from '../../store/ChrisStore';
import ErrorBoundary from '../ErrorBoundary/ErrorBoundary';
import './App.css';

// import the patternfly CSS globally
import '../../../node_modules/@patternfly/patternfly/patternfly.min.css';
import '../../../node_modules/@patternfly/patternfly/patternfly-base.css';
import '../../../node_modules/@patternfly/patternfly/patternfly-addons.css';
import '../../../node_modules/@patternfly/patternfly/patternfly-no-reset.css';

/*
 * The router here serves pages which replace the entire document,
 * whereas '../Router/Router' has a navbar and footer.
 */

const App = () => (
  <ChrisStore.Container>
    <ErrorBoundary>
      <div className="App">
        <BrowserRouter>
          <Switch>
            <Route path="/signin" component={ConnectedSignIn} />
            <Route path="/" component={Router} />
          </Switch>
        </BrowserRouter>
      </div>
    </ErrorBoundary>
  </ChrisStore.Container>
);

export default App;
