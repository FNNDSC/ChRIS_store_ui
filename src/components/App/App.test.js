import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a div with className App', () => {
    expect(wrapper.find('div.App')).toHaveLength(1);
  });

  it('should render a BrowserRouter component', () => {
    expect(wrapper.find('BrowserRouter')).toHaveLength(1);
  });

  it('should render a Switch component inside BrowserRouters', () => {
    expect(wrapper
      .find('BrowserRouter')
      .find('Switch'))
      .toHaveLength(1);
  });

  it('should render 2 routes inside Switch component', () => {
    expect(wrapper
      .find('Switch')
      .find('Route'))
      .toHaveLength(2);
  });
});
