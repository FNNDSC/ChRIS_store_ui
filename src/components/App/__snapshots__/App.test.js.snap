// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="App"
>
        <BrowserRouter>
                <Switch>
                        <Route
                                component={[Function]}
                                path="/signin"
                        />
                        <Route
                                component={[Function]}
                                path="/"
                        />
                </Switch>
        </BrowserRouter>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <BrowserRouter>
          <Switch>
                    <Route
                              component={[Function]}
                              path="/signin"
                    />
                    <Route
                              component={[Function]}
                              path="/"
                    />
          </Switch>
</BrowserRouter>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Switch>
            <Route
                        component={[Function]}
                        path="/signin"
            />
            <Route
                        component={[Function]}
                        path="/"
            />
</Switch>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/signin"
/>,
              <Route
                component={[Function]}
                path="/"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/signin",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="App"
>
          <BrowserRouter>
                    <Switch>
                              <Route
                                        component={[Function]}
                                        path="/signin"
                              />
                              <Route
                                        component={[Function]}
                                        path="/"
                              />
                    </Switch>
          </BrowserRouter>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <BrowserRouter>
            <Switch>
                        <Route
                                    component={[Function]}
                                    path="/signin"
                        />
                        <Route
                                    component={[Function]}
                                    path="/"
                        />
            </Switch>
</BrowserRouter>,
          "className": "App",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Switch>
              <Route
                            component={[Function]}
                            path="/signin"
              />
              <Route
                            component={[Function]}
                            path="/"
              />
</Switch>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  path="/signin"
/>,
                <Route
                  component={[Function]}
                  path="/"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/signin",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
